CONTRIBUTING.md

```diff
@@ -0,0 +1,505 @@
+- The `update_status` job makes sure the `ua status` command will have the latest
+information even when executed by a non-root user, updating the
+`/var/lib/ubuntu-advantage/status.json` file.
+
+The timer intervals can be changed using the `ua config set` command.
+```bash
+# Make the update_status job run hourly
+$ sudo ua config set update_status_timer=3600
+```
+Setting an interval to zero disables the job.
+```bash
+# Disable the update_status job
+$ sudo ua config set update_status_timer=0
+```
+
+## Directory layout
+The following describes the intent of UA client related directories:
+
+
+| File/Directory | Intent |
+| -------- | -------- |
+| ./tools | Helpful scripts used to publish, release or test various aspects of UA client |
+| ./features/ | Behave BDD integration tests for UA Client
+| ./uaclient/ | collection of python modules which will be packaged into ubuntu-advantage-tools package to deliver the UA Client CLI |
+| uaclient.entitlements | Service-specific \*Entitlement class definitions which perform enable, disable, status, and entitlement operations etc. All classes derive from base.py:UAEntitlement and many derive from repo.py:RepoEntitlement |
+| ./uaclient/cli.py | The entry-point for the command-line client
+| ./uaclient/clouds/ | Cloud-platform detection logic used in Ubuntu Pro to determine if a given should be auto-attached to a contract |
+| uaclient.contract | Module for interacting with the Contract Server API |
+| ./demo | Various stale developer scripts for setting up one-off demo environments. (Not needed often)
+| ./apt-hook/ | the C++ apt-hook delivering MOTD and apt command notifications about UA support services |
+| ./apt-conf.d/ | apt config files delivered to /etc/apt/apt-conf.d to automatically allow unattended upgrades of ESM security-related components. If apt proxy settings are configured, an additional apt config file will be placed here to configure the apt proxy. |
+| /etc/ubuntu-advantage/uaclient.conf | Configuration file for the UA client.|
+| /var/lib/ubuntu-advantage/private | `root` read-only directory containing Contract API responses, machine-tokens and service credentials |
+| /var/log/ubuntu-advantage.log | `root` read-only log of ubuntu-advantage operations |
+
+
+## Collecting logs
+The `ua collect-logs` command creates a tarball with all relevant data for debugging possible problems with UA. It puts together:
+- The UA Client configuration file (the default is `/etc/ubuntu-advantage/uaclient.conf`)
+- The UA Client log files (the default is `/var/log/ubuntu-advantage*`)
+- The files in `/etc/apt/sources.list.d/*` related to UA
+- Output of `systemctl status` for the UA Client related services
+- Status of the timer jobs, `canonical-livepatch`, and the systemd timers
+- Output of `cloud-id`, `dmesg` and `journalctl`
+
+Files with sensitive data are not included in the tarball. As of now, the command must be run as root.
```

Comment:
We include the files in the tarball, I think we just redact the sensitive information